src/solvers/flattening/boolbv.cpp

```diff
@@ -630,22 +630,11 @@ bool boolbvt::boolbv_set_equality_to_true(const equal_exprt &expr)
 
 void boolbvt::set_to(const exprt &expr, bool value)
 {
-  if(expr.type().id()!=ID_bool)
-  {
-    error() << "boolbvt::set_to got non-boolean operand: "
-            << expr.pretty() << eom;
-    throw 0;
-  }
-
-  if(value)
-  {
-    if(expr.id()==ID_equal)
-    {
-      if(!boolbv_set_equality_to_true(to_equal_expr(expr)))
-        return;
-    }
-  }
+  PRECONDITION(expr.type().id() == ID_bool);
 
+  const auto equal_expr = expr_try_dynamic_cast<equal_exprt>(expr);
+  if(value && equal_expr && !boolbv_set_equality_to_true(*equal_expr))
+    return;
   return SUB::set_to(expr, value);
```

Comment:
Nit pick: the `return` is misleading as `SUB::set_to` does not return a value. Just remove the `return` keyword. At which point the entire code should probably be:
```
if(!value || !equal_expr || boolbv_set_equality_to_true(*equal_expr))
  SUB::set_to(expr, value);
```

Reply:
I think @martin-cs was saying this is more difficult to understand, and I can agree as it has more negations, so I'm just going to remove the return